File: test/net/RobotRaconteurNETTest/RRAssert.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using RobotRaconteur;
using System.IO;
using System.Threading.Tasks;
using System.Runtime.CompilerServices;

namespace RobotRaconteurNETTest
{
static class RRAssert
{
    public static void AreEqual<T>(T a, T b, [CallerFilePath] string sourceFilePath = "",
                                   [CallerLineNumber] int sourceLineNumber = 0)
        where T : IComparable, IComparable<T>
    {
        if (a.CompareTo(b) != 0)
        {
            Console.WriteLine("Failure: {0} does not equal {1} at {2}:{3}", a, b, sourceFilePath, sourceLineNumber);
            throw new Exception("Unit test failure");
        }
    }

    public static void AreEqual(object a, object b, [CallerFilePath] string sourceFilePath = "",
                                [CallerLineNumber] int sourceLineNumber = 0)
    {
        if (!object.Equals(a, b))
        {
            Console.WriteLine("Failure: {0} does not equal {1} at {2}:{3}", a, b, sourceFilePath, sourceLineNumber);
            throw new Exception("Unit test failure");
        }
    }

    public static void AreNotEqual(object a, object b, [CallerFilePath] string sourceFilePath = "",
                                   [CallerLineNumber] int sourceLineNumber = 0)
    {
        if (object.Equals(a, b))
        {
            Console.WriteLine("Failure: {0} equals {1} at {2}:{3}", a, b, sourceFilePath, sourceLineNumber);
            throw new Exception("Unit test failure");
        }
    }

    public static void Fail([CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0)
    {
        Console.WriteLine("Failure: at {0}:{1}", sourceFilePath, sourceLineNumber);
        throw new Exception("Unit test failure");
    }

    public static void IsTrue(bool val, [CallerFilePath] string sourceFilePath = "",
                              [CallerLineNumber] int sourceLineNumber = 0)
    {
        if (!val)
        {
            Console.WriteLine("Failure: {0} is not true {1}:{2}", val, sourceFilePath, sourceLineNumber);
            throw new Exception("Unit test failure");
        }
    }

    public static void IsFalse(bool val, [CallerFilePath] string sourceFilePath = "",
                               [CallerLineNumber] int sourceLineNumber = 0)
    {
        if (val)
        {
            Console.WriteLine("Failure: {0} is not false {1}:{2}", val, sourceFilePath, sourceLineNumber);
            throw new Exception("Unit test failure");
        }
    }

    public static void ThrowsException<T>(Action f, [CallerFilePath] string sourceFilePath = "",
                                          [CallerLineNumber] int sourceLineNumber = 0)
        where T : Exception
    {
        bool thrown = false;
        try
        {
            f();
        }
        catch (T)
        {
            thrown = true;
        }
        if (!thrown)
        {
            Console.WriteLine("Failure: does not throw at {0}:{1}", sourceFilePath, sourceLineNumber);
            throw new Exception("Unit test failure");
        }
    }

    public static async Task ThrowsExceptionAsync<T>(Func<Task> f, [CallerFilePath] string sourceFilePath = "",
                                                     [CallerLineNumber] int sourceLineNumber = 0)
        where T : Exception
    {
        bool thrown = false;
        try
        {
            await f();
        }
        catch (T)
        {
            thrown = true;
        }
        if (!thrown)
        {
            Console.WriteLine("Failure: does not throw at {0}:{1}", sourceFilePath, sourceLineNumber);
            throw new Exception("Unit test failure");
        }
    }
}
}

File: test/java/test/com/robotraconteur/javatest/RRAssert.java
package com.robotraconteur.javatest;

class RRAssert
{

    public static void areEqual(byte a, byte b)
    {
        if (a != b)
        {
            System.out.println("Unit test failure: " + String.valueOf(a) + " is not equal to " + String.valueOf(b));
            throw new RuntimeException("Unit test failure");
        }
    }

    public static void areEqual(short a, short b)
    {
        if (a != b)
        {
            System.out.println("Unit test failure: " + String.valueOf(a) + " is not equal to " + String.valueOf(b));
            throw new RuntimeException("Unit test failure");
        }
    }

    public static void areEqual(int a, int b)
    {
        if (a != b)
        {
            System.out.println("Unit test failure: " + String.valueOf(a) + " is not equal to " + String.valueOf(b));
            throw new RuntimeException("Unit test failure");
        }
    }

    public static void areEqual(long a, long b)
    {
        if (a != b)
        {
            System.out.println("Unit test failure: " + String.valueOf(a) + " is not equal to " + String.valueOf(b));
            throw new RuntimeException("Unit test failure");
        }
    }

    public static void areEqual(Object a, Object b)
    {
        if (a == null && b == null)
        {
            return;
        }
        if (a == null || b == null)
        {
            System.out.println("Unit test failure: " + String.valueOf(a) + " is not equal to " + String.valueOf(b));
            throw new RuntimeException("Unit test failure");
        }
        if (!a.equals(b))
        {
            System.out.println("Unit test failure: " + String.valueOf(a) + " is not equal to " + String.valueOf(b));
            throw new RuntimeException("Unit test failure");
        }
    }

    public static void areNotEqual(Object a, Object b)
    {
        if (a == null && b == null)
        {
            System.out.println("Unit test failure: " + String.valueOf(a) + " is equal to " + String.valueOf(b));
            throw new RuntimeException("Unit test failure");
        }
        if (a == null || b == null)
        {
            return;
        }
        if (a.equals(b))
        {
            System.out.println("Unit test failure: " + String.valueOf(a) + " is equal to " + String.valueOf(b));
            throw new RuntimeException("Unit test failure");
        }
    }

    public static void areNotEqual(byte a, byte b)
    {
        if (a == b)
        {
            System.out.println("Unit test failure: " + String.valueOf(a) + " is equal to " + String.valueOf(b));
            throw new RuntimeException("Unit test failure");
        }
    }

    public static void areNotEqual(short a, short b)
    {
        if (a == b)
        {
            System.out.println("Unit test failure: " + String.valueOf(a) + " is equal to " + String.valueOf(b));
            throw new RuntimeException("Unit test failure");
        }
    }

    public static void areNotEqual(int a, int b)
    {
        if (a == b)
        {
            System.out.println("Unit test failure: " + String.valueOf(a) + " is equal to " + String.valueOf(b));
            throw new RuntimeException("Unit test failure");
        }
    }

    public static void areNotEqual(long a, long b)
    {
        if (a == b)
        {
            System.out.println("Unit test failure: " + String.valueOf(a) + " is equal to " + String.valueOf(b));
            throw new RuntimeException("Unit test failure");
        }
    }

    public static void fail()
    {
        throw new RuntimeException("Unit test failure");
    }

    public static void isTrue(boolean val)
    {
        if (!val)
        {
            throw new RuntimeException("Unit test failure");
        }
    }

    public static void isFalse(boolean val)
    {
        if (val)
        {
            throw new RuntimeException("Unit test failure");
        }
    }
}